src/Containers/RestaurantsList.js

```diff
@@ -0,0 +1,79 @@
+import React,{ Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {fetchRestaurants} from '../Actions/Actions';
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import Restaurant from "../Components/Restaurant";
+
+
+class RestaurantsList extends Component{
+
+    componentDidMount() {
+       this.props.getRestaurants();
+    }
+
+    render(){
+        const { restaurants, isFetching, lastUpdated, error } = this.props;
+        return(
+            <div>
+                <p>
+                    { lastUpdated && (
+                        <span>
+                            Last updated at {new Date(lastUpdated).toLocaleTimeString()}.{' '}
+                        </span>
+                    ) }
+                    { !isFetching && (
+                        <button onClick={() => this.props.getRestaurants()}>Refresh</button>
+                    )}
+                    {
+                        error && (
+                            <span>
+                                {error.message}
+                            </span>
+                        )
+                    }
+                </p>
+                <List>
+                    {restaurants.map(restaurant =>
+                        <ListItem key={restaurant.id} alignItems='flex-start'>
+                            <Restaurant restaurant={restaurant}/>
+                        </ListItem>
+                    )}
+                </List>
+            </div>
+
+        )
+    }
+
+}
+
+RestaurantsList.propTypes = {
+    restaurants: PropTypes.array.isRequired,
+    isFetching: PropTypes.bool.isRequired,
+    lastUpdated: PropTypes.number,
+    error: PropTypes.object,
+    getRestaurants: PropTypes.func.isRequired,
+};
+
+function mapStateToProps(state) {
+    const restaurants = state.items || [];
+    const isFetching = state.isFetching ||  false;
```

Comment:
why isn't there a default for `isFetching`?
also, pay attention that if `isFetching = false`, this "OR" condition will still execute (and evaluate to `false`)